core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -337,33 +337,16 @@ private[spark] object Utils extends Logging {
 
   def copyFileStreamNIO(
       input: FileChannel,
-      output: FileChannel,
+      output: WritableByteChannel,
       startPosition: Long,
       bytesToCopy: Long): Unit = {
-    val initialPos = output.position()
     var count = 0L
     // In case transferTo method transferred less data than we have required.
     while (count < bytesToCopy) {
       count += input.transferTo(count + startPosition, bytesToCopy - count, output)
     }
     assert(count == bytesToCopy,
       s"request to copy $bytesToCopy bytes, but actually copied $count bytes.")
-
-    // Check the position after transferTo loop to see if it is in the right position and
-    // give user information if not.
-    // Position will not be increased to the expected length after calling transferTo in
-    // kernel version 2.6.32, this issue can be seen in
-    // https://bugs.openjdk.java.net/browse/JDK-7052359
-    // This will lead to stream corruption issue when using sort-based shuffle (SPARK-3948).
-    val finalPos = output.position()
-    val expectedPos = initialPos + bytesToCopy
-    assert(finalPos == expectedPos,
-      s"""
-         |Current position $finalPos do not equal to expected position $expectedPos
-         |after transferTo, please check your kernel version to see if it is 2.6.32,
-         |this is a kernel bug which will lead to unexpected behavior when using transferTo.
-         |You can set spark.file.transferTo = false to disable this NIO feature.
-           """.stripMargin)
```

Comment:
read the links, and it seems to be a kernel bug? is this safe to delete?

Reply:
There is no position in writeableByteChannels :/ soo... idk ... meep

Reply:
If we keep this check it should be in the implementation of `ShuffleMapOutputWriter` I would guess. Maybe in `close`?

Reply:
Perhaps i'm missing something, but how do you get the expected position in the `ShuffleMapOutputWriter` for the assert? If that's not easy to get, I would say it's better to keep this here with a if-statement to check if it's an instance of `FileChannel` to maintain the same behavior as before.